net/modemmanager/files/modemmanager.proto

```diff
@@ -393,11 +394,16 @@ proto_modemmanager_setup() {
 		cliauth="${cliauth}${cliauth:+|}$auth"
 	done
 
+	if [ -n "$plmn" ]; then
+		echo "registering to network '${plmn}'..."
+		mmcli --modem="${device}" --3gpp-register-in-operator="$plmn"
```

Comment:
Just using the `operator-id` in the list of settings sent to `--simple-connect` should be enough; this explicit `--3gpp-register-in-operator` action should not be required. Plus, we're not checking the return status of this command, and that would mean that if it fails to register in that operator, it would still go on with the connection attempt.

The other option would be to add the error checking in the `--3gpp-register-in-operator` command, plus maybe add `--timeout=300` as the process may take longer than the default 30s timeout of a DBus request, and in that case, if the network registration to the explicit PLMN is done out of the connection attempt, then we could remove the `operator-id` setting in the `--simple-connect`.

In the very specific case of openwrt, where we configure the modem for data connectivity, I think using the `operator-id` only in the `simple-connect` should be more than enough.

Reply:
> Just using the operator-id in the list of settings sent to --simple-connect should be enough; this explicit --3gpp-register-in-operator action should not be required. Plus, we're not checking the return status of this command, and that would mean that if it fails to register in that operator, it would still go on with the connection attempt.

Have tried that but it doesn't register to requested network immediately - it still registers to home network  with bad signal.  
Afterwards if i do ifdown/up it registers to requested network.
Doing `--3gpp-register-in-operator` before connect works always.

Will check for error and unset `$plmn` if it fails

Reply:
Updated the patch.. Have added following in the commit description:

    The mmcli --3gpp-register-in-operator cmd always fails with
    "error: modem not enabled yet".
    
    This seems useless as "operator-id=MCC:MNC" is already passed in
    mmcli --simple-connect command.
    
    In my case not doing 3gpp-register-in-operator sometimes connects to
    wrong (home) MCC:MNC first time after reboot.

Reply:
> The mmcli --3gpp-register-in-operator cmd always fails with "error: modem not enabled yet".

This is telling you the command is fully ignored by ModemManager. It's like not sending the command at all.

> This seems useless as "operator-id=MCC:MNC" is already passed in mmcli --simple-connect command.

So, this is what is making the operator-id selection happen.

> In my case not doing 3gpp-register-in-operator sometimes connects to wrong (home) MCC:MNC first time after reboot.

That looks like a bug somewhere. Can you gather ModemManager debug logs when this happens, and report the issue in fd.o gitlab? https://gitlab.freedesktop.org/mobile-broadband/ModemManager/-/issues

So, after those comments, I'm now fully convinced we should remove the `--3gpp-register-in-operator` command from this patch.

Reply:
great... let me know if you'd like me to squash two commits.
I've tried rebooting the modem few times now and was unable to make it happen, it happened once before that.
As soon as it happens will open an upstream issue.

Reply:
It's been awhile, since this happened. If it happens it happens very rarely. :-/